homeassistant/components/mqtt/client.py

```diff
@@ -126,7 +133,9 @@ async def async_publish(
                 )
                 return
 
-    await hass.data[DATA_MQTT].async_publish(topic, outgoing_payload, qos, retain)
+    await mqtt_data.client.async_publish(
+        topic, outgoing_payload, qos or 0, retain or False
+    )
 
 
 AsyncDeprecatedMessageCallbackType = Callable[
```

Comment:
It doesn't cause any issues here, but could you correct the return type from `Awaitable[None]` to `Coroutine[Any, Any, None]`? That could prevent additional issues in the future.
https://github.com/home-assistant/core/blob/bbf54e6f443c4184bbf216d7e31edfb1f0171653/homeassistant/components/mqtt/client.py#L132-L134